packages/grafana-ui/src/components/LoadingBar/LoadingBar.mdx

```diff
@@ -0,0 +1,22 @@
+import { Meta, Preview, ArgsTable } from '@storybook/addon-docs/blocks';
+import { LoadingBar } from './LoadingBar';
+
+<Meta title="MDX|LoadingBar" component={LoadingBar} />
+
+# LoadingBar
+
+The LoadingBar is used as a simple loading slider animation in the top of its container.
+
+<Preview>
+  <div style={{ height: '200px', width: '400px' }}>
+    <LoadingBar containerWidth={400} height={2} width={128} />
+  </div>
+</Preview>
+
+```jsx
+<div style={{ height: '200px', width: '400px' }}>
+  <LoadingBar containerWidth={400} height={2} width={128} />
```

Comment:
Just a DX idea, I am not sure from the top of my head how hard it would be to implement it in a stable way:
Could we omit the `containerWidth` property and let the LoadingBar figure out the width of the parent container element? That could make the API simpler.

Reply:
> P.S. What's a DX?

Developer experience

Reply:
> seems too intrusive to me. What if the parent element has a display: flex and the LoadingBar is only 1 of its children, being allowed a space of 20% for example?

Let's don't overthink it :)

> However, it does seem like creating a wrapper inside the LoadingBar that has a width of 100% when a containerWidth is not set (or maybe even substituting it entirely) would work, as suggested by @Clarity-89 🙌.

I agree with @Clarity-89 about the sizing. IMO the loader should just have plain width: 100%, and the actual width control should happen in the parent (with i.e. additional wrapper, that can handle both width and layout - then the flex scenario described above is easy-peasy).

Reply:
Hi @dprokop, please check it out now - We added a wrapper inside the LoadingBar. The wrapper has width: 100% which would be as much as the parent allows for the LoadingBar component. The bar itself is inside the wrapper and its width can be set from the outside, but the whole wrapper is being animated because that way the bar successfully goes to the end of the allowed space.

Reply:
Sorry, I am a bit late on this :)

But yeah, basically what you discussed in thread is cool I think 👍